src/VTKBookLaTex/Chapter04.tex

```diff
@@ -2,164 +2,837 @@ \chapter{The Visualization Pipeline}
+
+\subsubsection{File Interface (Importers / Exporters)}
+\label{subsubsec:file_interface_importers_exporters}
+
+\emph{Importers} and \emph{exporters} are objects in the system that read or write data files consisting of more than one object. Typically importers and exporters are used to save or restore an entire scene (i.e., lights, cameras, actors, data, transformations, etc.). When an importer is executed, it reads one or more files and may create several objects. For exam-ple, in VTK the vtk3DSImporter imports a \emph{3D Studio} file and creates a rendering window, renderer, lights, cameras, and actors. Similarly, the vtkVRMLExporter creates a VRML file given a VTK render window. The VRML file contains cameras, lights, actors, geometry, transformations, and the like, indirectly referred to by the rendering window provided.
+
+In the \emph{Visualization Toolkit}, there are several importers and exporters. To see which importers and exporters are supported by VTK, see \emph{The VTK User's Guide}. You may also want to check the Web pages at \href{https://www.vtk.org}{https://www.vtk.org} for the current Doxygen manual pages. If the exporter you are looking for does not exist, you will have to develop your own using the programming interface.
+
+\begin{figure}[htb]
+  \begin{subfigure}[h]{0.48\linewidth}
+    \includegraphics[width=0.96\linewidth]{Figure4-13a}
+    \caption*{}
+    \label{fig:Figure4-13a}
+  \end{subfigure}
+  \hfill
+  \begin{subfigure}[h]{0.48\linewidth}
+    \includegraphics[width=0.96\linewidth]{Figure4-13b}
+    \caption*{)
+    \label{fig:Figure4-13b}
+  \end{subfigure}
+  \hfill
+  \begin{subfigure}[h]{0.96\linewidth}{Figure4-13c}
+    \caption*{}
+  \end{subfigure}
+  \caption{Importing and exporting files in VTK. An importer creates a vtkRenderWindow that describes the scene. Exporters use an instance of vtkRenderWindow to obtain a description of the scene. See \href{https://lorensen.github.io/VTKExamples/site/Cxx/IO/3DSImporter/}{3DSImporter.cxx} and \href{https://lorensen.github.io/VTKExamples/site/Python/IO/3DSImporter/}{3DSImporter.py}.}\label{fig:Figure4-13}
+\end{figure}
+
+Figure \ref{fig:Figure4-13} shows an image created from a *3D Studio* model and saved as a *Renderman* RIB file.
+
+\subsection{Application Interface}
+\label{subsec:application_interface}
+
+The majority of users interface to their data by using an existing application. Rather than programming pipelines or writing their own readers and writers, users acquire an application that suits their particular visualization needs. Then to interface to their data, users simply id
```

Comment:
So I included back the mention to the `LoopShrk.cxx` example. I guess it should point [here](https://gitlab.kitware.com/vtk/vtk/blob/3ad5fa14aaa30c57cab400a7494bd6e2eac37bf1/graphics/examplesCxx/LoopShrk.cxx), but the markdown version does not have an hyperlink either, so I'll wait until @lorensen can tell what to do for this case.